From 22b7c7c00e01a7534787940f206b5e78a2f788eb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Mon, 1 Nov 2021 18:09:33 +0100
Subject: [PATCH 02/21] xen: handle CONFIG_STUBDOM

Adapted from https://github.com/QubesOS/qubes-vmm-xen-stubdom-linux/blob/2a814bd1edaf549ef9252eb6747aa6137abf9831/qemu/patches/qemu-xen-h.patch

Authors: HW42

Contributors: fepitre
---
 include/hw/xen/xen.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/hw/xen/xen.h b/include/hw/xen/xen.h
index 37ecc91fc3..8a16ec951e 100644
--- a/include/hw/xen/xen.h
+++ b/include/hw/xen/xen.h
@@ -37,6 +37,15 @@ extern uint32_t xen_domid;
 extern enum xen_mode xen_mode;
 extern bool xen_domid_restrict;
 
+static inline int xen_stubdom_enable(void)
+{
+#ifdef CONFIG_STUBDOM
+    return 1;
+#else
+    return 0;
+#endif
+}
+
 int xen_pci_slot_get_pirq(PCIDevice *pci_dev, int irq_num);
 int xen_set_pci_link_route(uint8_t link, uint8_t irq);
 void xen_intx_set_irq(void *opaque, int irq_num, int level);
-- 
2.41.0

